module/assets/helper.js

```diff
@@ -0,0 +1,416 @@
+
+        return resolve({
+            result: organizationAssets,
+          });
+        
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+
+  /**
+   * Transfer ownership Kafka Event.
+   * @method
+   * @name ownershipTransfer
+   * @param {ownershipTransferEvent} - OwenerShip transfer Event message object 
+   * {
+      "highWaterOffset": 63,
+       "key": "",
+       "offset": 62,
+       "partition": 0,
+       "topic": "ownershiptransfer",
+       "edata": {
+          "action": "ownership-transfer",
+          "organisationId": "01269934121990553633",
+          "context": "Ownership Transfer",
+        "actionBy": {
+             "userId": "86d2d978-5b20-4453-8a76-82b5a4c728c9",
+            "userName": ""
+            },
+      "fromUserProfile": {
+              "userId": "19d81ef7-36ce-41fe-ae2d-c8365d977be4",
+              "userName": "",
+              "channel": "",
+              "organisationId": "",
+              "roles": [
+                 "PROGRAM_MANAGER"
+                 ]
+             },
+       "toUserProfile": {
+              "userId": "86d2d978-5b20-4453-8a76-82b5a4c728c9",
+              "userName": "test",
+              "firstName": "test",
+              "lastName": "",
+           "roles": [
+                "PROGRAM_MANAGER",
+                "CONTENT_CREATOR"
+               ]
+            },
+       "assetInformation": {
+             "objectType": "solution || program",
+             "identifier": "{{resource_identifier}}"
+           },
+         "iteration": 1
+  }
+}
+   * @returns {Promise} success Data.
+   */
+  static ownershipTransfer(ownershipTransferEvent) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let reqData = ownershipTransferEvent.edata;
+        let fromFindQuery = {
+          userId: reqData.fromUserProfile.userId,
+          platformRoles: { $exists: true },
+        };
+
+        let updateUserSolutionsDataResult;
+        let checkUsersRolesIsIdentical= assetService.checkRolesPresence(
+          reqData.fromUserProfile.roles,
+          reqData.fromUserProfile.roles
+        );
+
+        if (
+          reqData.actionBy.userId &&
+          reqData.fromUserProfile.userId &&
+          reqData.toUserProfile.userId && checkUsersRolesIsIdentical
+        ) {
+          //get Fromuser details from user Extension
+
+          let fromUserData = await userExtensionsHelper.userExtensionDocument(
+            fromFindQuery
+          );
+          //get Touser details from user Extension
+          let toFindQuery = {
+            userId: reqData.toUserProfile.userId,
+          };
+          let toUserData = await userExtensionsHelper.userExtensionDocument(
+            toFindQuery
+          );
+          let allAssetsData = fromUserData?.platformRoles;
+          let typeOfAssetsToMove = reqData.assetInformation.objectType;
+          let checkAssetInformation =
+            reqData.hasOwnProperty("assetInformation");
```

Comment:
put this check first and then do the rest operation

Reply:
resolved